File: frontend/src/services/profile-service.ts
import { httpClient } from '../lib';
import type { ChangePasswordRequest, ProfileResponse, ProfileUpdateRequest } from '../types';

export async function getProfile() {
  const response = await httpClient.get<ProfileResponse>('/profile');
  return response.data;
}

export async function updateProfile(data: ProfileUpdateRequest) {
  const response = await httpClient.put<ProfileResponse>('/profile', data);
  return response.data;
}

export async function changePassword(data: ChangePasswordRequest) {
  const response = await httpClient.post<ProfileResponse>('/profile/change-password', data);
  return response.data;
}

export async function uploadAvatar(file: File) {
  const formData = new FormData();
  formData.append('file', file);
  const response = await httpClient.post<ProfileResponse>('/profile/avatar', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return response.data;
}

File: backend/src/main/java/project/lms_rikkei_edu/modules/profile/service/ProfileService.java
package project.lms_rikkei_edu.modules.profile.service;

import org.springframework.web.multipart.MultipartFile;
import project.lms_rikkei_edu.modules.profile.dto.request.ChangePasswordRequest;
import project.lms_rikkei_edu.modules.profile.dto.request.ProfileUpdateRequest;
import project.lms_rikkei_edu.modules.profile.dto.response.ProfileResponse;

import java.util.UUID;

public interface ProfileService {

    ProfileResponse getProfile(UUID userId);

    ProfileResponse updateProfile(UUID userId, ProfileUpdateRequest request);

    ProfileResponse changePassword(UUID userId, ChangePasswordRequest request);

    ProfileResponse uploadAvatar(UUID userId, MultipartFile file);
}
